### PR TITLE
Improve types in js sdk

### PR DESCRIPTION
**Pull request by apstanisic**
Specify return types of methods

### CODE REVIEW

File: packages/sdk-js/src/handlers/server.ts [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+		project_logo: string | null;
+		project_color: string | null;
+		public_foreground: string | null;
+		public_background: string | null;
+		public_note: string | null;
+		custom_css: null;
+	};
+};
+
+export type OasData = {
```
Comment (rijkvanzanten) [inline packages/sdk-js/src/handlers/server.ts:15]: We should be able to reuse types from `openapi3-ts` in here

File: packages/sdk-js/src/handlers/server.ts [hunk 2]
```diff
@@ -1,5 +1,30 @@
 import { AxiosInstance } from 'axios';
 
+export type ServerInfo = {
```
Comment (rijkvanzanten) [inline packages/sdk-js/src/handlers/server.ts:3]: There's more info in here when you're authenticated

**Review by rijkvanzanten — COMMENTED** [review]
Nice!

You can use the generic `Response` type to wrap stuff in that `{ data: business }`:

https://github.com/directus/directus/blob/426d4c1a105c0a356f6250d213a42e9e1fe60be1/packages/sdk-js/src/types.ts#L10-L13

File: packages/sdk-js/src/handlers/server.ts (hunk 2 above)
Comment (apstanisic) [reply packages/sdk-js/src/handlers/server.ts:3]: added

File: packages/sdk-js/src/handlers/server.ts (hunk 1 above)
Comment (apstanisic) [reply packages/sdk-js/src/handlers/server.ts]: Yeah, I didn't want to add dependency without checking with you. Added

**Comment by apstanisic** [comment]
> Nice!
> 
> You can use the generic `Response` type to wrap stuff in that `{ data: business }`:
> 
> https://github.com/directus/directus/blob/426d4c1a105c0a356f6250d213a42e9e1fe60be1/packages/sdk-js/src/types.ts#L10-L13

Does it matter that none of return types have meta object, and that are never null?

Should I do something like this bellow, or to simply replace `{data: Type}` with `Response<Type>`?
```typescript
export interface Response<T> {
	data: T
}

export interface ItemsResponse<T> extends Response<T | null> {
	data: T | null;
	meta?: Record<Meta, number>;
}
```

**Comment by rijkvanzanten** [comment]
Ohh that's why you went with just `{ data: T }`, my bad! I thought you might've missed the Response type 🙂

File: packages/sdk-js/package.json [hunk 3]
```diff
@@ -27,6 +27,7 @@
   "devDependencies": {
     "chai": "^4.2.0",
     "mocha": "^8.2.0",
+    "openapi3-ts": "^2.0.0",
```
Comment (rijkvanzanten) [inline packages/sdk-js/package.json:30]: No need to add it to devDependencies here. It's pulled in from the root package.json (monorepo's ftw)

File: packages/sdk-js/src/handlers/server.ts [hunk 4]
```diff
@@ -1,4 +1,20 @@
...
+		project_logo: string | null;
+		project_color: string | null;
+		public_foreground: string | null;
+		public_background: string | null;
+		public_note: string | null;
+		custom_css: null;
+	};
+	directus: { version: string };
+	node: { uptime: number; version: string };
+	os: { totalmem: number; type: string; uptime: number; version: string };
```
Comment (rijkvanzanten) [inline packages/sdk-js/src/handlers/server.ts:16]: These will have to be optional, as they won't always be returned

File: packages/sdk-js/package.json (hunk 3 above)
Comment (ialameera) [reply packages/sdk-js/package.json:30]: saad74657@gmail.com

Comment (apstanisic) [reply packages/sdk-js/package.json:30]: Awesome!

When the user installs standalone sdk, he/she won't have `openapi3-ts` in `node_modules`, or am I missing something?
I think we either need to either add it to `dependencies`, embed `OpenApiObject` in project or simply return `any`?

This is `server.d.ts`
```typescript
import { AxiosInstance } from 'axios';
import type { OpenAPIObject } from 'openapi3-ts';
export declare type ServerInfo = {
    project: {
        project_name: string;
        project_logo: string | null;
        project_color: string | null;
        public_foreground: string | null;
        public_background: string | null;
        public_note: string | null;
        custom_css: null;
    };
    directus?: {
        version: string;
    };
    node?: {
        uptime: number;
        version: string;
    };
    os?: {
        totalmem: number;
        type: string;
        uptime: number;
        version: string;
    };
};
export declare class ServerHandler {
    private axios;
    constructor(axios: AxiosInstance);
    specs: {
        oas: () => Promise<OpenAPIObject>;
    };
    ping(): Promise<'pong'>;
    info(): Promise<{
        data: ServerInfo;
    }>;
}
```

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**